app/questionnaire/questionnaire_schema.py

```diff
@@ -400,11 +398,11 @@ def _is_list_name_in_rule(
                 )
 
             # Nested rules
-            if any(operator in rule for operator in OPERATION_MAPPING):
+            if has_operator(rule):
                 return self._is_list_name_in_rule(rule, list_name)
 
     @staticmethod
-    def get_operands(rules: Mapping) -> list:
+    def get_operands(rules: Union[Mapping, Sequence]) -> list:
```

Comment:
How come this changed?. When would this be a Sequence, as the code below would fail if this was?

Reply:
I can't seem to get around this one, not sure why it thinks there is a Sequence being passed to this method, but I get `Argument 1 to "get_operands" of "QuestionnaireSchema" has incompatible type "Union[Mapping[Any, Any], Sequence[Any]]"; expected "Mapping[Any, Any]"` in mypy

Reply:
Mypy can't work out this would only get called with a Mapping. But this method shouldn't change because it technically couldn't be a sequence anyway as it would fall over. 

I think for this reason, we should put the `isinstance` check inline as well which would fix the issue, i.e: 
```py
if isinstance(rules, Mapping) and has_operator(rules):
   do_stuff()
```

Reply:
Updated this 👍